engine/schema/src/com/cloud/storage/dao/StoragePoolTagsDaoImpl.java

```diff
@@ -77,4 +90,71 @@ public void deleteTags(long poolId) {
         txn.commit();
     }
 
+    @Override
+    public List<StoragePoolTagVO> searchByIds(Long... stIds) {
+        String batchCfg = _configDao.getValue("detail.batch.query.size");
```

Comment:
@nvazquez what about extracting lines 95-97 to a method?
Then, we can have test cases and some doc for it. Especially for that 2000 magic number there.